### ISSUE TITLE
Linker errors when building benchmarks with MSVC on Windows

### ISSUE THREAD

**Issue by felipecrv** [apache/arrow#34203]
### Describe the bug, including details regarding any error messages, version, and platform.

Without `-DBENCHMARK_STATIC_DEFINE` the Google Benchmark library declares symbols as `__declspec(dllexport)` and that confuses the linker when `benchmark::benchmark` is statically linked into our benchmarks.

Google Benchmark itself gets built with the proper defines but binaries depending on it including the header without passing the correct define tell the linker to not look for the symbol in the static library.

The clean CMake fix is to add the define to the interface library target:

```cmake
target_compile_definitions(benchmark::benchmark INTERFACE BENCHMARK_STATIC_DEFINE)
```

### Component(s)

C++

**Comment by felipecrv** [comment]
take

**Comment by felipecrv** [comment]
Closing as this is a duplicate of #34188 that is fixed by #34194
